mmv1/third_party/terraform/main.go.erb

```diff
@@ -2,11 +2,71 @@
 package main
 
 import (
-	"github.com/hashicorp/terraform-plugin-sdk/v2/plugin"
+	"context"
+	"flag"
+
+	"github.com/hashicorp/terraform-plugin-framework/providerserver"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-go/tfprotov5"
+	"github.com/hashicorp/terraform-plugin-mux/tf5muxserver"
+	"github.com/hashicorp/terraform-plugin-mux/tf6to5server"
 	"github.com/hashicorp/terraform-provider-google<%= "-" + version unless version == 'ga'  -%>/google<%= "-" + version unless version == 'ga'  -%>"
 )
 
+var (
+	// these will be set by the goreleaser configuration
+	// to appropriate values for the compiled binary
+	version string = "dev"
+
+	// goreleaser can also pass the specific commit if you want
+	// commit  string = ""
+)
+
 func main() {
-	plugin.Serve(&plugin.ServeOpts{
-		ProviderFunc: google.Provider})
+	var debug bool
+
+	flag.BoolVar(&debug, "debug", false, "set to true to run the provider with support for debuggers like delve")
+	flag.Parse()
+
+	// plugin framework provider
+	downgradedFrameworkProvider, err := tf6to5server.DowngradeServer(
+		context.Background(),
+		providerserver.NewProtocol6(google.New(version)),
```

Comment:
Drive-by comment: Rather than serving the framework provider via protocol version 6 then downgrading to protocol version 5, the mux server can use the framework provider on protocol version 5 directly instead: https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/providerserver#NewProtocol5

```go
providers := []func() tfprotov5.ProviderServer{
	providerserver.NewProtocol5(google.New(version)),
	google.Provider().GRPCProvider,
}
```